lib/unleash/configuration.rb

```diff
@@ -40,7 +39,9 @@ def validate!
       return if self.disable_client
 
       raise ArgumentError, "URL and app_name are required parameters." if self.app_name.nil? || self.url.nil?
-      raise ArgumentError, "custom_http_headers must be a hash." unless self.custom_http_headers.is_a?(Hash)
+      unless self.custom_http_headers.is_a?(Hash) || self.custom_http_headers.respond_to?(:call)
+        raise ArgumentError, "custom_http_headers must be a Hash or a Proc."
+      end
```

Comment:
:cry: 

Good point. Maybe we can rename `ensure_valid_opts(opts)` into `valid_custom_http_headers?(custom_http_headers)` and move the logic there, so the line here becomes shorter?

```ruby
   def valid_custom_http_headers?(custom_http_headers)
      custom_http_headers.is_a?(Hash) || custom_http_headers.respond_to?(:call)
   end
```

It could then be called from both here, and `ensure_valid_opts()` is originally called from as `ensure_valid_opts(self.custom_http_headers)`

Reply:
Loved it!